agent/utils/utils_windows.go

```diff
@@ -80,3 +82,18 @@ func IsAvailableDriveLetter(hostPath string) bool {
 	}
 	return false
 }
+
+func MkdirAllAndChown(path string, perm fs.FileMode, uid, gid int) error {
```

Comment:
Why is this method not common? I mean do we need platform specific file?
The internal methods it calls are common for both Linux and Windows?